Fleet UI: Add timestamps to host count on software detail pages

RachelElysia opened the pull request:
## Issue
For #24975

## Description
- Add last updated at counts for the software details page, os details page, and vulnerabilities details page
- These counts are all baseline
- Fixed all last updated counts to not be bolded

## Screenshot of all 3
<img width="1240" alt="Screenshot 2025-01-06 at 12 02 47 PM" src="https://github.com/user-attachments/assets/aa24f526-19a9-47e4-bf44-b0d84df32ca8" />
<img width="924" alt="Screenshot 2025-01-06 at 12 02 29 PM" src="https://github.com/user-attachments/assets/514926ef-bdd4-4b41-b54b-333b7d387fc2" />
<img width="1254" alt="Screenshot 2025-01-06 at 12 02 15 PM" src="https://github.com/user-attachments/assets/cb74c82d-be3e-4c67-888d-ac252ad9ac40" />


# Checklist for submitter

If some of the following don't apply, delete the relevant line.

<!-- Note that API documentation changes are now addressed by the product design team. -->

- [x] Changes file added for user-visible changes in `changes/`, `orbit/changes/` or 
- [x] Manual QA for all new/changed functionality

iansltx reviewed (commented):
Code looks fine (though it feels like we could extract the host count tooltip message to a constant somewhere), but styling is a bit off. Which probably means tweaking the last-updated counts element.

<img width="296" alt="host count styling" src="https://github.com/user-attachments/assets/d82d1f2a-8a91-4b07-ac93-5d2ee646517b" />

Checked the font on that item in Figma and it's showing bold rather than standard, and I'm 99% sure the font-weight on the added items isn't bold. Additionally, seems like we're missing a tiny bit of `left-pad`ing on the count element to match up with Figma (and it feels like we do need that padding). Can't eyeball this exactly (I probably could if host count was 48) but maybe 4-8px?

RachelElysia commented:
@iansltx thanks for catching these, I just did what the default was for current styling

- All other updated at isn't bold, I updated this PR to match figma bold. @eugkuo is this intentional?
- Added in the 8px gap
- Aligned baseline to match Figma, though other updated at isn't aligned baseline. @eugkuo is this intentional

Example of other updated at text not being aligned baseline and not being bold:
<img width="319" alt="Screenshot 2025-01-06 at 10 57 59 AM" src="https://github.com/user-attachments/assets/96e80e6f-5e2f-4af0-8449-791676edb2d5" />

eugkuo commented:
@RachelElysia 

> All other updated at isn't bold, I updated this PR to match figma bold. @eugkuo is this intentional?

Looking at the live software pages it looks like "Updated" is bold: 
1. https://dogfood.fleetdm.com/software/titles?team_id=0
2. https://dogfood.fleetdm.com/software/os?team_id=0&page=0
3. https://dogfood.fleetdm.com/software/vulnerabilities?team_id=0&page=0

Which is probably why I made it bold here. That said, I don't think it needs to/should be as, as you point out, it isn't elsewhere in the platform. I def like it better unbolded. I'll make the changes to unbold in the associated figma file. Thanks for calling that out!

>Aligned baseline to match Figma, though other updated at isn't aligned baseline. @eugkuo is this intentional

It's actually been annoying me that text isn't baseline aligned in the platform. I'm baseline aligning it on the pages I touch and def would like to see it baseline aligned throughout going forward. It looks a little like a mistake (or that the tooltip underline is pushing the text up) when it's not baseline aligned imho.

RachelElysia commented:
> It looks a little like a mistake (or that the tooltip underline is pushing the text up) when it's not baseline aligned imho.

@eugkuo - sweet, I agree. Maybe let's make a bug ticket for this so we can track fixing it everywhere. I'll push it aligned baseline here to get that ball rolling.

I'll unbold the text and push up to this PR

RachelElysia commented on frontend/components/LastUpdatedText/_styles.scss:
```diff
@@ -1,5 +1,6 @@
 .component__last-updated-text {
   font-size: $xx-small;
+  font-weight: $regular;
```
Updated globally here

jacobshandling replied:
Thanks Rachel. Can you also please remove [these lines](https://github.com/fleetdm/fleet/blob/aef4bb579ed1dfbaddc648b5e4bfca2bef975011/frontend/pages/policies/ManagePoliciesPage/_styles.scss#L299-L301) (same fix, locally) with merging this global update

eugkuo commented:
> @eugkuo - sweet, I agree. Maybe let's make a bug ticket for this so we can track fixing it everywhere. I'll push it aligned baseline here to get that ball rolling.

@RachelElysia I've created this bug ticket. I'll populate it with figma links in the "To fix" section tonight: https://github.com/fleetdm/fleet/issues/25180

RachelElysia commented:
@iansltx ready for ur review <3

iansltx reviewed (approved):
`LastUpdatedHostCount` feels a little bit weird naming-wise, vs. e.g. `HostCountWithFreshness`, but my opinion on this isn't strong enough to block. Otherwise code LGTM; thanks for the cleanup/componentization!
